encodings/dict/src/dict.rs

```diff
@@ -90,12 +89,13 @@ impl ArrayValidity for DictArray {
                 .into_primitive()
                 .vortex_expect("Failed to convert DictArray codes to primitive array");
             match_each_integer_ptype!(primitive_codes.ptype(), |$P| {
-                ArrayAccessor::<$P>::with_iterator(&primitive_codes, |iter| {
-                    LogicalValidity::Array(
-                        BoolArray::from(iter.flatten().map(|c| *c != 0).collect::<Vec<_>>())
-                            .into_array(),
-                    )
-                }).vortex_expect("Failed to convert DictArray codes into logical validity")
+                let is_valid = primitive_codes
+                    .maybe_null_slice::<$P>()
+                    .iter()
+                    .map(|c| * c != 0)
```

Comment:
formatting here is funky, you can make it `copied()` to remove `*c`

Reply:
lol I missed this. good catch

Reply:
rust-analyzer refuses to format in macros for some reason